pkg/private/tar/build_tar.py

```diff
@@ -273,31 +273,48 @@ def add_tree(self, tree_top, destpath, mode=None, ids=None, names=None):
           to_write[dest_dir + file] = content_path
 
     for path in sorted(to_write.keys()):
+      if file_attributes:
+          attrs = file_attributes(path, with_default_mode=False)
+          use_ids = attrs.get("ids")
+          use_names = attrs.get("names")
+          # in this case, use_mode will be None and mode (as passed into this
+          # function) will be the default_mode.
+          use_mode = attrs.get("mode")
+      else:
+          use_ids = ids
+          use_names = names
+          use_mode = mode
+
       content_path = to_write[path]
       if not content_path:
         # This is an intermediate directory. Bazel has no API to specify modes
         # for this, so the least surprising thing we can do is make it the
-        # canonical rwxr-xr-x
+        # canonical rwxr-xr-x unless we've gotten a custom override
+        if use_mode is None:
```

Comment:
It seems like just having 0o755 be the default mode would eliminate the need for the test here.